### ISSUE TITLE
⌨ VoiceOver bug on Diversity, equity and inclusion page

### ISSUE THREAD

**Issue by Elisa0122** [GCTC-NTGC/gc-digital-talent#6113]
## ❌ Current behaviour (issue)
Overview of current behaviour.

**WCAG 2.1 Criteria:**

https://user-images.githubusercontent.com/83783101/228653808-d302c3d6-b8c2-4eaf-9437-9dd290ec712f.mov

### How to reproduce
1. Environment is UAT
2. Which browser and/or Assistive Technology? Chrome and Apple VoiceOver
3. Step-by-step instructions explaining how to reproduce the problem. Navigate to **Diversity, equity and inclusion** page (`/users/:userId/profile/employment-equity/edit`) and start VoiceOver and navigate through the items

## ✔ Expected behaviour (fix)
Read the link only once and remove the hidden text about the collapse button.

### Acceptance Criteria
A set of assumptions which, when tested, verify that the accessibility requirement was met.

- [ ] VoiceOver reads the link once
- [ ] VoiceOver does not  read random text about the collapse button

## Additional information (optional)
Eric mention this maybe related to the #4868 
![Screenshot 2023-03-29 at 4 07 30 PM](https://user-images.githubusercontent.com/83783101/228655255-cbbaee21-752d-4009-8354-eaaa03122d29.png)

**Comment by Elisa0122** [comment]
Hi Peter,
Thank you for taking over the bug,
Sorry for the delay in the reply.
I spoke to Josh and he will reply to you.
@substrae

**Comment by substrae** [comment]
@petertgiles @Elisa0122 

Sorry for the delay on this, I started a reply before leaving the office and then promptly forgot to finish it. 🤦‍♂️

> I know many of the devs would like to see us move away from hidden text because of the issues it creates, like this.

Unfortunately I don't see this going away. The need for this text arose from the fact that AT offers the ability to read a page's links/buttons out in a list, removed from their context in the interface. This means that this page would read out something like: "Remove from profile. Add to profile. Add to profile. Add to profile."

While the visual context of the page element enables relational context for sighted users, we know that's not useful for AT users, so we're presented with two options:
- Adjust the link label to be universally more descriptive so that context is always clear
- Provide an enhanced experience for AT users that enhances their specific context

The goal here is to make the call based on creating a good experience for both groups, and while it might be ideal to strive for one consistent experience for everyone, doing so harms both experiences unnecessarily. In this specific case, in order to create universal parity for both groups, we'd have to literally repeat the block's label in the link text, meaning you end up with something like this:

![image](https://user-images.githubusercontent.com/6960386/230407300-444d24e7-1874-4a35-a729-8be8dfd9dba9.png)

This visual experience has been significantly downgraded in terms of usability both in the sense of legibility and cognitive load.

Using hidden text here is helpful because it allows us to create a clear experience for both groups without impacting either negatively. Solving the hidden text problem should be a one-and-done effort, so once it works in VoiceOver (as I understand it @esizer?) we'll have a reusable tool for other instances where we can't repeat information in close proximity.

**Comment by esizer** [comment]
> Using hidden text here is helpful because it allows us to create a clear experience for both groups without impacting either negatively. Solving the hidden text problem should be a one-and-done effort, so once it works in VoiceOver (as I understand it @esizer?) we'll have a reusable tool for other instances where we can't repeat information in close proximity.

Yeah, I was actually as a solution (because Apple is not usually quick to fix any webkit bugs 😓 ) was to make the entire area a button rather than restricting it to the right side and essentially duplicating the text in a hidden spot for AT users. 

So, instead of reading like "Woman, add woman to my profile" it would just read "Add woman to my profile" or something else that reads a bit better 😛 .

**Comment by RM-1978** [comment]
@esizer and @substrae  Thank you for your thoughts.  I agree with @esizer that the plainer language versions might be a better user experience for those of use who listen with Screen readers.

**Comment by esizer** [comment]
I believe this was addressed with the new application and profile pages.
